File: sdks/thunkmetrc-go/wrapper/models/item_category.go
package models

type ItemCategory struct {
    CanBeDecontaminated bool `json:"CanBeDecontaminated,omitempty"`
    CanBeDestroyed bool `json:"CanBeDestroyed,omitempty"`
    CanBeRemediated bool `json:"CanBeRemediated,omitempty"`
    CanContainSeeds bool `json:"CanContainSeeds,omitempty"`
    LabTestBatchNames []interface{} `json:"LabTestBatchNames,omitempty"`
    Name string `json:"Name,omitempty"`
    ProductCategoryType string `json:"ProductCategoryType,omitempty"`
    QuantityType string `json:"QuantityType,omitempty"`
    RequiresAdministrationMethod bool `json:"RequiresAdministrationMethod,omitempty"`
    RequiresAllergens bool `json:"RequiresAllergens,omitempty"`
    RequiresDescription bool `json:"RequiresDescription,omitempty"`
    RequiresItemBrand bool `json:"RequiresItemBrand,omitempty"`
    RequiresLabelPhotoDescription bool `json:"RequiresLabelPhotoDescription,omitempty"`
    RequiresLabelPhotos int64 `json:"RequiresLabelPhotos,omitempty"`
    RequiresNumberOfDoses bool `json:"RequiresNumberOfDoses,omitempty"`
    RequiresPackagingPhotoDescription bool `json:"RequiresPackagingPhotoDescription,omitempty"`
    RequiresPackagingPhotos int64 `json:"RequiresPackagingPhotos,omitempty"`
    RequiresProductPDFDocuments int64 `json:"RequiresProductPDFDocuments,omitempty"`
    RequiresProductPhotoDescription bool `json:"RequiresProductPhotoDescription,omitempty"`
    RequiresProductPhotos int64 `json:"RequiresProductPhotos,omitempty"`
    RequiresPublicIngredients bool `json:"RequiresPublicIngredients,omitempty"`
    RequiresServingSize bool `json:"RequiresServingSize,omitempty"`
    RequiresStrain bool `json:"RequiresStrain,omitempty"`
    RequiresSupplyDurationDays bool `json:"RequiresSupplyDurationDays,omitempty"`
    RequiresUnitCbdAContent bool `json:"RequiresUnitCbdAContent,omitempty"`
    RequiresUnitCbdAContentDose bool `json:"RequiresUnitCbdAContentDose,omitempty"`
    RequiresUnitCbdAPercent bool `json:"RequiresUnitCbdAPercent,omitempty"`
    RequiresUnitCbdContent bool `json:"RequiresUnitCbdContent,omitempty"`
    RequiresUnitCbdContentDose bool `json:"RequiresUnitCbdContentDose,omitempty"`
    RequiresUnitCbdPercent bool `json:"RequiresUnitCbdPercent,omitempty"`
    RequiresUnitThcAContent bool `json:"RequiresUnitThcAContent,omitempty"`
    RequiresUnitThcAContentDose bool `json:"RequiresUnitThcAContentDose,omitempty"`
    RequiresUnitThcAPercent bool `json:"RequiresUnitThcAPercent,omitempty"`
    RequiresUnitThcContent bool `json:"RequiresUnitThcContent,omitempty"`
    RequiresUnitThcContentDose bool `json:"RequiresUnitThcContentDose,omitempty"`
    RequiresUnitThcPercent bool `json:"RequiresUnitThcPercent,omitempty"`
    RequiresUnitVolume bool `json:"RequiresUnitVolume,omitempty"`
    RequiresUnitWeight bool `json:"RequiresUnitWeight,omitempty"`
}

File: sdks/thunkmetrc-cs/ThunkMetrc.Wrapper/Models/ItemCategory.cs
using System;
using System.Collections.Generic;

namespace ThunkMetrc.Wrapper.Models
{
    public class ItemCategory
    {
        [global::System.Text.Json.Serialization.JsonPropertyName("CanBeDecontaminated")]
        public required bool CanBeDecontaminated { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("CanBeDestroyed")]
        public required bool CanBeDestroyed { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("CanBeRemediated")]
        public required bool CanBeRemediated { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("CanContainSeeds")]
        public required bool CanContainSeeds { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("LabTestBatchNames")]
        public required List<object> LabTestBatchNames { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("Name")]
        public required string Name { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ProductCategoryType")]
        public required string ProductCategoryType { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("QuantityType")]
        public required string QuantityType { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresAdministrationMethod")]
        public required bool RequiresAdministrationMethod { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresAllergens")]
        public required bool RequiresAllergens { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresDescription")]
        public required bool RequiresDescription { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresItemBrand")]
        public required bool RequiresItemBrand { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresLabelPhotoDescription")]
        public required bool RequiresLabelPhotoDescription { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresLabelPhotos")]
        public required int RequiresLabelPhotos { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresNumberOfDoses")]
        public required bool RequiresNumberOfDoses { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresPackagingPhotoDescription")]
        public required bool RequiresPackagingPhotoDescription { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresPackagingPhotos")]
        public required int RequiresPackagingPhotos { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresProductPDFDocuments")]
        public required int RequiresProductPDFDocuments { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresProductPhotoDescription")]
        public required bool RequiresProductPhotoDescription { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresProductPhotos")]
        public required int RequiresProductPhotos { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresPublicIngredients")]
        public required bool RequiresPublicIngredients { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresServingSize")]
        public required bool RequiresServingSize { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresStrain")]
        public required bool RequiresStrain { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresSupplyDurationDays")]
        public required bool RequiresSupplyDurationDays { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitCbdAContent")]
        public required bool RequiresUnitCbdAContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitCbdAContentDose")]
        public required bool RequiresUnitCbdAContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitCbdAPercent")]
        public required bool RequiresUnitCbdAPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitCbdContent")]
        public required bool RequiresUnitCbdContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitCbdContentDose")]
        public required bool RequiresUnitCbdContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitCbdPercent")]
        public required bool RequiresUnitCbdPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitThcAContent")]
        public required bool RequiresUnitThcAContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitThcAContentDose")]
        public required bool RequiresUnitThcAContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitThcAPercent")]
        public required bool RequiresUnitThcAPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitThcContent")]
        public required bool RequiresUnitThcContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitThcContentDose")]
        public required bool RequiresUnitThcContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitThcPercent")]
        public required bool RequiresUnitThcPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitVolume")]
        public required bool RequiresUnitVolume { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RequiresUnitWeight")]
        public required bool RequiresUnitWeight { get; set; }
    }
}
